programs/sol-xr/src/instructions/open_minting_round.rs

```diff
@@ -0,0 +1,157 @@
+use crate::mint_round::{AssociatedRoundAccount, MintRound};
+use crate::{Invest, InvestBumps};
+use anchor_spl::metadata::Metadata;
+use std::ops::Div;
+use {
+    crate::state::sol_strategy::SolStrategy,
+    anchor_lang::prelude::Rent,
+    anchor_lang::prelude::*,
+    anchor_lang::system_program,
+    anchor_spl::{
+        associated_token::AssociatedToken,
+        token::{mint_to, Mint, MintTo, Token, TokenAccount},
+    },
+};
+
+#[derive(Accounts)]
+#[instruction(id: u64,market_value: u64)]
+pub struct OpenMintingRound<'info> {
+    #[account(
+        mut,
+        constraint = sol_strategy.governance_authority.key() == governance_authority.key() @ Error::UnauthorizedGovernanceAuthority,
+    )]
+    pub governance_authority: Signer<'info>,
+
+    #[account(
+        mut,
+        seeds = [b"token"],
+        bump
+    )]
+    pub token: Account<'info, Mint>,
+
+    #[account(
+        mut,
+        seeds = [SolStrategy::SEED_PREFIX],
+        bump
+    )]
+    pub sol_strategy: Account<'info, SolStrategy>,
+
+    #[account(
+        init_if_needed,
+        payer = governance_authority,
+        constraint = sol_strategy.allow_new_mint == false @ Error::MintingAlreadyAllowed,
+        constraint = sol_strategy.next_minting_rounds == id @ Error::IncorrectRoundId,
+        constraint = sol_strategy.next_minting_rounds <= sol_strategy.minting_rounds @ Error::AllMintRoundsCompleted, // todo: test after multiple round creation
+        constraint = token.supply > 0 @ Error::TokenSupplyIsZero,
+        space = 8 + MintRound::INIT_SPACE,
+        seeds = [MintRound::SEED_PREFIX,&id.to_le_bytes()],
+        bump
+    )]
+    pub mint_round: Account<'info, MintRound>,
+
+    pub token_program: Program<'info, Token>,
+    pub associated_token_program: Program<'info, AssociatedToken>,
+    pub system_program: Program<'info, System>,
+}
+
+impl<'info> OpenMintingRound<'info> {
+    pub fn handler(
+        &mut self,
+        bumps: &OpenMintingRoundBumps,
+        id: u64,
+        market_value: u64,
+    ) -> Result<()> {
+        let pass_mint_value_requirement = Self::check_mint_value_requirement(
+            market_value,
+            self.sol_strategy.min_premium_nav_ratio,
+            self.sol_strategy.sol_in_treasury,
+            self.token.supply,
+        )?;
+        require!(pass_mint_value_requirement, Error::MarketValueBelowMinPremium);
+
+        self.sol_strategy.allow_new_mint = true;
+
+        self.mint_round.premium = market_value;
+        self.mint_round.max_mint_per_wallet = self.sol_strategy.max_mint_per_wallet;
```

Comment:
`max_mint_per_wallet` should be a fraction of `sol_xr` available to mint. Ideally 5-10%